crates/nu-command/src/network/port.rs

```diff
@@ -87,15 +87,17 @@ fn get_free_port(
         }
 
         // try given port one by one.
-        let addrs: Vec<SocketAddr> = (start_port..=end_port)
-            .map(|current| {
-                SocketAddr::V4(SocketAddrV4::new(
-                    Ipv4Addr::new(127, 0, 0, 1),
-                    current as u16,
-                ))
-            })
-            .collect();
-        TcpListener::bind(addrs.as_slice())?
+        match (start_port..=end_port)
+            .map(|port| SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, port as u16)))
```

Comment:
Mhh the [truncating `as u16` coercion](https://doc.rust-lang.org/stable/reference/expressions/operator-expr.html#numeric-cast) is still there, so this seems to fix another part of error handling but doesn't prevent the overflow itself? Maybe worth migrating to `try_into` or explicit checking beforehand that it is in the valid range of a port.

Reply:
This PR fixes  `Vec<SocketAddr>` overflow.  Casting `port` cast to `u16` is another issue. I think we could check `port`  explicitly.

Reply:
I created a pr #11214.